gulpfile.js

```diff
@@ -1,41 +1,83 @@
 const gulp = require('gulp')
-const autoprefixer = require('gulp-autoprefixer')
 const changed = require('gulp-changed')
-const stylus = require('gulp-stylus')
 const browserSync = require('browser-sync').create()
 const childProcesses = require('child_process')
 const chalk = require('chalk')
 const fs = require('fs')
+const path = require('path')
+const webpack = require('webpack')
+const webpackStream = require('webpack-stream')
+const ManifestPlugin = require('webpack-manifest-plugin')
+const ExtractTextPlugin = require('extract-text-webpack-plugin')
 
 // Define a global serverProcess that we can use to keep track of the running server instance
 let serverProcess = null
 
+// Whether our webpack task should be watching or not
+let isStaticBuildWatching = false
+
+const PROJECT_ROOT = __dirname;
 const EXEC_PATH = './.build/debug/HaCWebsite'
+const isProduction = process.env.NODE_ENV === 'production'
 
-// A straight copy of files that don't require processing
-// You should update this glob if you ever add processing to a new file type
-function copy () {
-  return gulp.src('static/src/**/!(*.styl)')
-    .pipe(changed('static/dist'))
-    .pipe(gulp.dest('static/dist'))
+function setWatchOnBuildStatic () {
+  isStaticBuildWatching = true
+  return Promise.resolve()
 }
 
-// Process the Stylus files into CSS
-function buildStyles () {
-  return gulp.src('static/src/styles/main.styl')
-    .pipe(changed('static/dist/styles'))
-    .pipe(stylus({
-      'include css': true,
-      paths: ['./node_modules']
-    }))
-    .on('error', function (err) {
-      console.log('[' + chalk.red('Stylus error...') + ']')
-      console.log(err.message)
-      console.log('\u0007')
-      this.emit('end')
-    })
-    .pipe(autoprefixer())
-    .pipe(gulp.dest('static/dist/styles'))
+function buildStatic () {
+  // To prevent lots of files and add ease to live reloading,
+  // We don't want to hash filenames in dev mode
+  const dotHash = isProduction ? '.[hash]' : ''
```

Comment:
What's this square bracket syntax?

Reply:
That's a webpack-ism: https://webpack.js.org/configuration/output/#output-filename